### ISSUE TITLE
Disable automatic check for updates

### ISSUE THREAD

**Issue by xinatanil** [overtake/TelegramSwift#206]
If you install Telegram for macOS via .dmg (not via Mac App Store), then you will frequently get those dialogs with "New version of Telegram is available, would you like to install it, etc". The problem is that minor releases of Telegram are being released on a very frequent basis, so this update pop up appears almost every day for me.

Pressing "Skip Update" works fine, but new releases come very often, and there's no way to disable automatic app update in the settings, so this pop up became INCREDIBLY annoying for me.

Please, add an option to disable app updates completely.

**Comment by ivan-verizon** [comment]
That's why I'm still using an old version. A new version has a huge issue with the updater

**Comment by Zipfer** [comment]
Is it possible to disable the update, because it downloads the updates in the background and spends the internet tethering traffic?
Moreover, it redownloads the update each time telegram restarted

**Comment by Tom5om** [comment]
I agree, give us the option to not install updates. I am travelling so essentially always on my 4g data. There should be a checkbox to disable this

**Comment by FerryA1** [comment]
Agreed, the option should be in Preferences.

For example, using Telegram in an old machine (macOS 10.11 "El Capitan", cannot be upgraded, does not accept newer OS versions), and last update (9.0 STABLE) does not work on it, keeps crashing at start, logging out and back in as suggested does not work. Only reinstalling the old version (8.8 STABLE, fortunately I keep it stored) makes it work again, but it keeps DL'ing update and popping up a button to "Update" in the middle of the channel list, lower side, not too difficult to click it inadvertently, so a "Don't check for updates" option would be nice.

**Comment by 1111Al** [comment]
> found the answer myself go to users/urname/Library/Caches/ru.keepcoder.Telegram/org.sparkle-project.sparkle
> 
> make the folder "PersistentDownloads" read only and lock

It doesn't wotk, that is the permission for users not for apps. Anyway, after i set that folder on read only, after a while I found that folder permission was reverse to read and write permission, and the usual two auto update files had been downloaded in there.

**Comment by jiangtann** [comment]
> ⚠️ eat with caution
> 
> ```
> cd ~/Library/Caches/ru.keepcoder.Telegram/org.sparkle-project.Sparkle/
> sudo mv PersistentDownloads PersistentDownloads.bak
> sudo install -d -g wheel -m 700 -o root PersistentDownloads
> ```

Every time I reopen Telegram, it automatically starts downloading update data. Although it ultimately results in the message "_Can't make a temporary directory for the update download at (null)_" and annoying "Update Telegram" button disappears, it still consumes my network traffic. Is there a way to prevent Telegram from automatically downloading these updates?

**Comment by AndreyGubin** [comment]
> Is there a way to prevent Telegram from automatically downloading these updates?

There is no way to prevent updates due stupid Sparkle framework. It uses telegram servers to download updates so I can't block it with hosts or brandmauer. I'm thinking to use different telegram client because of consuming traffic for useless updates

**Comment by d4rky-pl** [comment]
Considering that important bugs are not being fixed (#840, #1110), this is a must

**Comment by 1111Al** [comment]
The only working solution is to install Telegram app from Apple store, disabling automatic updates in the Apple Store preferences.

**Comment by 1111Al** [comment]
> > The only working solution is to install Telegram app from Apple store, disabling automatic updates in the Apple Store preferences.
> 
> The command line solution works:
> 
> [#206 (comment)](https://github.com/overtake/TelegramSwift/issues/206#issuecomment-1845533312)

I didn't try it 'cause I forgot this thread, but now with App Store version the problem is gone.

**Comment by IfSheDiesSheDies** [comment]
This button harasses me, I dont care to reopen apps every 5 seconds

**Comment by noideastowork** [comment]
Is there any way to download Telegram 10.1.4 (App Store version)? It has less bugs.

**Comment by notpushkin** [comment]
There’s also something wrong with the cache, at least for me:

```
% du -d0 -h ./Caches/ru.keepcoder.Telegram/org.sparkle-project.Sparkle/PersistentDownloads/
 13G	./Caches/ru.keepcoder.Telegram/org.sparkle-project.Sparkle/PersistentDownloads/
```

I’ve just cleaned it up and updated using brew, and it has downloaded the same version twice already.

**Comment by artk42** [comment]
Just noticed that updater is infinitely trying to download update:

<img width="230" height="126" alt="Image" src="https://github.com/user-attachments/assets/bf8cfee6-87b6-4e4c-84fa-06456782d4d1" />

Tried to find and block it via lulu it worked for 10 minutes but than switched to alternative ip and went crazy again. Unless a solution found to prevent it - be careful on using app counted traffic connections

**Comment by notpushkin** [comment]
I’ve skimmed over the code and I think it falls back to fetching updates from a Telegram channel (or something like that). So there is no way to block updates without blocking Telegram itself ¯\\\_(ツ)\_/¯ 

(which makes sense in case e.g. your ISP or government tries to block Telegram, you can then use a proxy and stay updated, but it also means the broken updater will keep burning through your metered traffic)
